src/qt/receiverequestdialog.cpp

```diff
@@ -183,9 +183,19 @@ void ReceiveRequestDialog::update()
             QPainter painter(&qrAddrImage);
             painter.drawImage(0, 0, qrImage.scaled(QR_IMAGE_SIZE, QR_IMAGE_SIZE));
             QFont font = GUIUtil::fixedPitchFont();
-            font.setPixelSize(12);
-            painter.setFont(font);
             QRect paddedRect = qrAddrImage.rect();
+
+            // calculate ideal font size
+            qreal fontSize = 14; // don't go larger then 14 points
```

Comment:
Please factor this out into a function: this makes it easier to read and I think this might come in useful in other places as well.